### PR TITLE
Converted Exp6 and Exp7 to Cairo 2.X

### PR DESCRIPTION
**Pull request by Nonnyjoe**
WIP ....... what are your reviews?

### CODE REVIEW

**Comment by Nonnyjoe** [comment]
WIP ....... what are your reviews?

File: src/ex07.cairo [hunk 1]
```diff
@@ -5,8 +5,13 @@
 // - Use this contract's claim_points() function
 // - Your points are credited by the contract
 ////////////////////////////////
+#[starknet::interface] 
+trait Ex07Trait<T> {
```
Comment (TosinJs) [inline src/ex07.cairo]: The first argument to your Trait functions should be a reference to the Contracts state as seen here.

https://book.starknet.io/chapter_2/functions.html#traits

File: src/ex07.cairo [hunk 2]
```diff
@@ -5,8 +5,13 @@
...
+    fn claim_points(value_a: u128, value_b: u128);
+    fn update_class_hash(class_hash: felt252);
+}
 
-#[contract]
+#[starknet::contract]
```
Comment (TosinJs) [inline src/ex07.cairo:14]: This contract does not have a storage value, ContractState cannot be accessed without this. If there are no storage values for this example I believe an empty storage value will suffice.

https://book.starknet.io/chapter_2/storage.html

Comment (Nonnyjoe) [reply src/ex07.cairo:14]: Thanks for the correction, I've implemented an empty storage value.

File: src/ex07.cairo (hunk 1 above)
Comment (Nonnyjoe) [reply src/ex07.cairo]: Thanks, I've adjusted the code accordingly

**Review by omarespejel — APPROVED** [review]

**Comment by omarespejel** [comment]
Hey @Nonnyjoe amazing work! Merging it now.

Thanks @TosinJs for your review.

**Pull request merged**